src/validator.js

```diff
@@ -221,10 +221,10 @@ export default class Validator {
           return this.detectTypeFromLayout()
             .then((addonType) => {
               addonMetaData.type = addonType;
-              return Promise.resolve(addonMetaData);
+              return addonMetaData;
```

Comment:
Hmm...

The `Promise.resolve()` isn't needed but it clears up what is eventually returned... and what happens if we chain something else after this promise--do we need to add `Promise.resolve()` back?